prov/mrail/src/mrail_init.c

```diff
@@ -52,9 +58,54 @@ static inline char **mrail_split_addr_strc(const char *addr_strc)
 
 static int mrail_parse_env_vars(void)
 {
+	char *str;
 	char *addr_strc;
+	int val;
 	int ret;
 
+	/* experimental, subject to change */
+	fi_param_define(&mrail_prov, "algorithm_small", FI_PARAM_STRING,
+			"Rail utilizing algorithm for small messages "
+			"(fixed[:<rail>], round-robin, striping, default:fixed)");
+	ret = fi_param_get_str(&mrail_prov, "algorithm_small", &str);
+	if (!ret) {
+		if (!strcasecmp(str, "striping")) {
+			mrail_algorithm_small = MRAIL_ALGORITHM_STRIPING;
+		} else if (!strcasecmp(str, "round-robin")) {
+			mrail_algorithm_small = MRAIL_ALGORITHM_ROUND_ROBIN;
+		} else if (!strcasecmp(str, "fixed")) {
+			mrail_algorithm_small = MRAIL_ALGORITHM_FIXED;
+		} else if (!strncasecmp(str, "fixed:", 6)) {
+			mrail_algorithm_small = MRAIL_ALGORITHM_FIXED;
+			mrail_default_rail_small = atoi(str + 6);
+		}
+	}
+
+	/* experimental, subject to change */
+	fi_param_define(&mrail_prov, "algorithm_large", FI_PARAM_STRING,
+			"Rail utilizing algorithm for large messages "
+			"(fixed[:<rail>], round-robin, striping, default:striping)");
+	ret = fi_param_get_str(&mrail_prov, "algorithm_large", &str);
+	if (!ret) {
+		if (!strcasecmp(str, "striping")) {
+			mrail_algorithm_large = MRAIL_ALGORITHM_STRIPING;
+		} else if (!strcasecmp(str, "round-robin")) {
+			mrail_algorithm_large = MRAIL_ALGORITHM_ROUND_ROBIN;
+		} else if (!strcasecmp(str, "fixed")) {
+			mrail_algorithm_large = MRAIL_ALGORITHM_FIXED;
+		} else if (!strncasecmp(str, "fixed:", 6)) {
+			mrail_algorithm_large = MRAIL_ALGORITHM_FIXED;
+			mrail_default_rail_large = atoi(str + 6);
+		}
+	}
+
+	/* experimental, subject to change */
+	fi_param_define(&mrail_prov, "max_size_small", FI_PARAM_INT,
+			"Maximum message size considered as small ");
+	ret = fi_param_get_int(&mrail_prov, "max_size_small", &val);
+	if (!ret)
+		mrail_max_size_small = val;
```

Comment:
I can envision the rail algorithm choices getting really complex, with sub-options differing based on which algorithm is selected.  Maybe a control file might be a better choice?

Reply:
A control file can be used in the future, when we have better picture about most of the needed options. These envars are good for now as an experimental measure.